pkg/pillar/pubsub/socketdriver/subscribe.go

```diff
@@ -381,26 +381,19 @@ func (s *Subscriber) translate(in <-chan string, out chan<- pubsub.Change) {
 	min := max / 3
 	ticker := flextimer.NewRangeTicker(time.Duration(1000*min),
 		time.Duration(1000*max))
-	ticker.StopTicker()
 
 	for {
 		select {
-		case e, ok := <-ticker.C:
-			if !ok {
-				// We get some spurious indications with e being zero
-				break
-			}
-
-			if !gotRestarted {
-				s.log.Fatalf("ticker without gotRestarted val %s, e %v",
-					restartedValue, e)
+		case _, ok := <-ticker.C:
+			if !ok || !gotRestarted {
+				// skip handling in case of closed or not restarted
+				continue
 			}
 			out <- pubsub.Change{Operation: pubsub.Restart, Key: restartedValue}
 			s.log.Functionf("sent restarted file with %s",
 				restartedValue)
 			gotRestarted = false
 			restartedValue = ""
-			ticker.StopTicker()
```

Comment:
Should really restart with the 1000x timer values instead of doing a Stop() here and elsewhere.

Reply:
Done, it will reduce ticks count.